python/cudf/cudf/_lib/io/datasource.pyx

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from cudf._lib.cpp.io.types cimport datasource
+
+cdef class Datasource:
+    cdef datasource* get_datasource(self) except +:
+        pass
```

Comment:
I think we want this to throw some type of Exception to make it clear it's an abstract base class not to be used directly.